src/solver/simulation/economy.h

```diff
@@ -47,7 +47,8 @@ class EconomyWeeklyOptimization
 {
 public:
     using Ptr = std::unique_ptr<EconomyWeeklyOptimization>;
-    virtual void solve(Variable::State& state, int hourInTheYear, uint numSpace) = 0;
+    virtual void solve(Variable::State& state, int hourInTheYear, uint numSpace, uint week) = 0;
+    virtual void solveCSR(const Variable::State& state, uint numSpace, uint week) = 0;
```

Comment:
We should not force the definition of **solveCSR** if it can be empty (in case of no adequacy patch).
**solve(...)** as the only pure virtual method is enough.